src/tools/illink/src/linker/Linker.Steps/MarkScopeStack.cs

```diff
@@ -78,6 +78,21 @@ public MarkScopeStack ()
 			_scopeStack = new Stack<Scope> ();
 		}
 
+		internal LocalScope PushLocalScope (in MessageOrigin origin)
```

Comment:
Why bother creating new functions instead of changing the signatures of the old ones?

Reply:
`PushScope` and `PopToParent` are on the public API, and I don't think we'd want to expose a type that used to be private.

Reply:
Oh, I see, I didn't realize this was a public API. Makes sense then.